laurent22/joplin
# Mobile: Confirm closing settings with unsaved changes

# Summary
This should help address confusion similar to that discussed [on the forum](https://discourse.joplinapp.org/t/how-to-synchonize-between-windows-10-and-android/28479/3).

~~This won't work correctly until #7503 is fixed. Thus, it might be reasonable to close this pull request until #7503 is resolved.~~

# Testing plan
1. Open settings
2. Press the back arrow without changing settings (no confirm dialog should appear).
3. Open settings again
4. Change a setting
5. Press the back arrow without clicking "save"
6. Click "discard changes" and confirm that changes were discarded.
7. Repeat steps 4 and 5.
8. Click "save changes" and confirm that changes were saved.
9. Open settings again
10. Change a setting and click "save"
11. Close settings. The confirm dialog should not be visible.
